htdocs/accountancy/journal/purchasesjournal.php

```diff
@@ -336,6 +345,31 @@
 		dol_print_error($db);
 	}
 }
+*/
+// New way, single query, load all unbound lines
```

Comment:
Good idea.
However something is missing. In previous call, we fill the array $errorforinvoice only for invoices into the array $tabfac.
With the new code, we save ton of SQL request, but if the number of invoice in database is very large, we may return a  very large number of data, even if user has a filter to work with only 10 invoices. Time saved may be transformed into not enough memory error.
So I recommend to add a test in the SQL
`AND fk_facture_fourn IN ('.$db->sanitize(join(',', array_keys($tabfac))).')';`

A good practice is to not have too much elements inside a IN of a SQL (mysql and postgresql has no limit because they creates a temporary table and make a join when nb is > 999).
But i suggest to introduce a limit, no existing for the moment
```
if (count($tabfac) > 10000) {
$error++;
setEventMessages("TooManyInvoicesToProcessPleaseUseAMoreSelectiveFilter", null, 'errors');
}

```

Reply:
Committed the changes.
Should `if (count($tabfac) > 10000) {` go as a config variable from `htdocs/admin/const.php`??

Reply:
This is a security against bad use.
100% of users should be ableto work with this limit of 10 000, so i see no reason for the moment to allow user to setup this.
A good thing would be however to reproduce the same optimization for the page salesjournal.php